### PR TITLE
[s] [Heavily WIP] Ports goonstation CID randomizer detector

### PR DESCRIPTION
**Pull request by Shadowlight213**
Still working on cleaning the code up and stuff. Not tested at all.
Mainly making the pr now so i can get help.
@MrStonedOne I'm going to need some help with what sql calls i'm going to need to get ckey cid lists and stuff.

### CODE REVIEW

File: code/__HELPERS/cidtest.dm [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+	save_compids(C.ckey, C.compid_info_list)
+
+var/global/list/cid_test = list()
+var/global/list/cid_tested = list()
+
+/proc/do_computerid_test(client/C)
+	var/cid = cid_test[C.ckey]
+	if(!cid) return //They were not scheduled for testing
+	var/is_fucker = cid != C.computer_id //IT CHANGED!!!
```
Comment (PKPenguin321) [inline code/__HELPERS/cidtest.dm]: nice var names

File: code/__HELPERS/cidtest.dm [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	SF.eof = -1
+	for(var/datum/compid_info/CI in cid_list)
+		SF << CI
+
+/proc/check_compid_list(var/client/C)
+	C.compid_info_list = load_compids(C.ckey)
+
+	var/append_CID = 1
+
+	for(var/datum/compid_info/CI in C.compid_info_list)
```
Comment (RemieRichards) [inline code/__HELPERS/cidtest.dm]: this list should be associative.

File: code/__HELPERS/cidtest.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/__HELPERS/cidtest.dm]: gooncode, grandfathered in

File: code/__HELPERS/cidtest.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/__HELPERS/cidtest.dm]: l o o p y
y e s
t y p e y

Comment (RemieRichards) [inline code/__HELPERS/cidtest.dm]: actually istypeless loop is also wrong goof, there should be NO loop, it should just be assoc.

**Comment by MrStonedOne** [comment]
You can get a player's last cid with 

`SELECT computerid from [format_table("player")] where ckey = [ckey]` but you would have to do it before save_client_to_db() in client/New() (or add it to save_client_to_db())

In fact, this system could be a lot more simpler then goon's because we already track last cid in the db. I'll just pr a quicky.

File: code/__HELPERS/cidtest.dm (hunk 2 above)
Comment (Iamgoofball) [inline code/__HELPERS/cidtest.dm]: loop this:
8=====D

**Pull request closed without merging**